docs/architecture-decisions/fractional-non-string-rand-units.md

```diff
@@ -0,0 +1,167 @@
+
+### 2. Same "value" (e.g. 57.2, "some text", true, etc) should result in the same bucket assignment no matter the language of the provider and platform used
+
+Please note:
+
+* some languages (e.g. Python) don't necessarily have standard types by default (e.g. int32 vs int64).
+* [OpenFeature spec 312](https://openfeature.dev/specification/sections/evaluation-context/#requirement-312) dictates that evaluation context needs to support `boolean` | `string` | `number` | `structure` | `datetime` types.
+* JSON supports 6 fundamental types: `boolean` | `string` | `number` | `object` | `array` | `null`
+
+As such, the encodings for the following types as first argument (either as literals or results of evaluation) will be standardized:
+
+1. boolean
+2. string
+3. integer (any integer number, Python style)
+4. float (any floating point number, Python style)
+5. object (structure / map)
+6. datetime
+7. null
+
+**array / sequence** will be explicitly not supported as the first argument in fractional so it's possible to distinguish between hashing input and variant bucket. Nevertheless, it can be a part of object type and its encoding needs to be standardized as well.
+
+## Non-requirements
+
+* This change does not need to be backward-compatible.
+* Support advanced features like salting non-string types in JSON directly (that will be a separate ADR).
+* Bucketing improvements (that will be a separate ADR).
+
+## Considered Options
+
+1. **Proposed:** *Type-Aware Hashing:* Extend the current behavior to support non-string types as first arguments to `fractional`.
+2. *New Operator:* Introduce a new operator, such as `"bytesVar"`, to explicitly signal that the variable's raw bytes should be hashed.
+3. *Operator Overloading:* Reuse an existing operator (e.g., `"merge"`) or structure (e.g., providing a list) to imply byte-based hashing.
+
+Option 1 was chosen for its ergonomics and zero-impact on existing schemas. Option 2 adds unnecessary complexity to the flag definition language, and Option 3 creates confusing and non-obvious semantics.
+
+## Proposal
+
+We will modify the evaluation logic for the `fractional` operator.
+
+When inspecting the first element of the `fractional` array:
+
+1. If the first element in `fractional` evaluates to a non-array type then deterministically encode it to a well defined byte array and hash the bytes.
+2. Otherwise, if `targetingKey` is a string, build a 2-elements array of `flagKey` and `targetingKey`, deterministically encode that and hash (**NOTE:** This is different than string concatenation used today).
+3. Otherwise, if `targetingKey` is non-string, report an error and return a default value (as this breaks the [OpenFeature spec](https://openfeature.dev/specification/glossary/#targeting-key)).
+4. Otherwise, if `targetingKey` is missing, report an error and return a default value.
```

Comment:
> return a default value.

I believe the fractional operator should evaluate to a falsy value like `null` if targetingKey is missing.